hybrid/decomposers.py

```diff
@@ -316,6 +318,59 @@ def next(self, state, **runopts):
         return state.updated(subproblem=subbqm)
 
 
+class RoofDualityDecomposer(Runnable, traits.ProblemDecomposer):
```

Comment:
Since new samples are produced, maybe we should mark it also with `traits.ProblemSampler` or `traits.SamplesProducing`.